src/test/java/org/mifos/integrationtest/cucumber/stepdef/VoucherManagementStepDef.java

```diff
@@ -295,12 +305,10 @@ public void createAndActivateVoucher() {
         mockServerStepDef.startStub("/createVoucher", PUT, 200);
         mockServerStepDef.startStub("/activateVoucher", PUT, 200);
         iCallTheVoucherCreateAPIWithExpectedStatusOf(202, "/createVoucher");
-        try {
-            Thread.sleep(5000);
-        } catch (InterruptedException e) {
-            throw new RuntimeException(e);
-        }
-        iShouldBeAbleToExtractResponseBodyFromCallback();
+        await().atMost(awaitMost, SECONDS).pollInterval(pollInterval, SECONDS).until(() -> {
+            iShouldBeAbleToExtractResponseBodyFromCallback();
```

Comment:
add assert statement

Reply:
Here it calls the extract response from callback and assert is present there.